lib/foreman/renderer/scope/macros/helpers.rb

```diff
@@ -11,6 +11,33 @@ module Helpers
             sections only: %w[all]
           end
 
+          apipie :method, "Returns counter increased by given value (as String) stored in a hostgroup param named 'sequence_num_xxx'" do
+            required :hostgroup_name, String, "hostgroup title to store in (e.g. RHEL7/base)"
+            optional :increment, Integer, "integer increment, can be negative too", default: 1
+            optional :name, String, "name of the sequence (prefix for hostgroup parameter)", default: 'default'
+            optional :prefix, Integer, "leading zeroes", default: 1
+            returns String, "String with incremented value or empty string if hostgroup not found"
+            example 'sequence_hostgroup_param_next("myHG", 10) #=> 10'
+            example 'sequence_hostgroup_param_next("myHG", 2, "kiosk", 5) #=> 00002'
+          end
+          def sequence_hostgroup_param_next(hostgroup_name, increment = 1, name = "default", prefix = 1)
```

Comment:
Could this be a hostgroup that isn't related to the host in question?
Perhaps this helper should be in `HostTemplate` and only allow accessing the current host's hostgroup?

Reply:
Yeah, as I explained - this was initially written for discovery where host is not available.

Reply:
I think that makes sense, whatever hostgroup you want to use as a counter